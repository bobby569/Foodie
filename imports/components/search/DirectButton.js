import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Button, Icon } from 'antd';

class DirectButton extends Component {
	constructor(props) {
		super(props);

		this.renderPage = this.renderPage.bind(this);
	}

	renderPage(e) {
		const { name } = e.target;
		this.props.renderPage(name);
	}

	render() {
		const { name } = this.props;
		return (
			<div className="div-center">
				<Button.Group>
					<Button
						name="lib"
						type="primary"
						ghost={name !== 'lib'}
						onClick={this.renderPage}
					>
						<Icon type="bulb" />Recipe Library
					</Button>
					<Button
						className="diy"
						name="diy"
						type="primary"
						ghost={name !== 'diy'}
						onClick={this.renderPage}
					>
						By My Recipe<Icon type="user" />
					</Button>
				</Button.Group>
			</div>
		);
	}
}

DirectButton.propTypes = {
	name: PropTypes.string.isRequired,
	renderPage: PropTypes.func.isRequired
};

export default DirectButton;
