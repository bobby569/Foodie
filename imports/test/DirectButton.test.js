import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import DirectButton from '../components/search/DirectButton';

const renderPage = jest.fn();

describe('DirectButton', () => {
	const direct_button = shallow(
		<DirectButton name="lib" renderPage={renderPage} />
	);
	const rendered_lib = renderer.create(
		<DirectButton name="lib" renderPage={renderPage} />
	);
	const rendered_diy = renderer.create(
		<DirectButton name="diy" renderPage={renderPage} />
	);

	it('Renders correctly', () => {
		expect(direct_button.exists()).toBe(true);
		expect(rendered_lib.toJSON()).toMatchSnapshot();
		expect(rendered_diy.toJSON()).toMatchSnapshot();
	});
});
