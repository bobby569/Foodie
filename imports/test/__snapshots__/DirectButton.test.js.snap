// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DirectButton Renders correctly 1`] = `
<div
  className="div-center"
>
  <div
    className="ant-btn-group"
  >
    <button
      className="ant-btn ant-btn-primary"
      name="lib"
      onClick={[Function]}
      type="button"
    >
      <i
        className="anticon anticon-bulb"
      />
      <span>
        Recipe Library
      </span>
    </button>
    <button
      className="ant-btn ant-btn-primary ant-btn-background-ghost"
      name="diy"
      onClick={[Function]}
      type="button"
    >
      <span>
        By My Recipe
      </span>
      <i
        className="anticon anticon-user"
      />
    </button>
  </div>
</div>
`;

exports[`DirectButton Renders correctly 2`] = `
<div
  className="div-center"
>
  <div
    className="ant-btn-group"
  >
    <button
      className="ant-btn ant-btn-primary ant-btn-background-ghost"
      name="lib"
      onClick={[Function]}
      type="button"
    >
      <i
        className="anticon anticon-bulb"
      />
      <span>
        Recipe Library
      </span>
    </button>
    <button
      className="ant-btn ant-btn-primary"
      name="diy"
      onClick={[Function]}
      type="button"
    >
      <span>
        By My Recipe
      </span>
      <i
        className="anticon anticon-user"
      />
    </button>
  </div>
</div>
`;
